### PR TITLE
amptk v1.2.0: fix sha256 and add DOI

### PR DESCRIPTION
**Pull request by nextgenusfs**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

Adding DOI to recipe.

### CODE REVIEW

File: recipes/amptk/meta.yaml [hunk 1]
```diff
@@ -8,10 +8,10 @@ package:
 source:
   fn: {{ name }}-{{ version }}.tar.gz
   url: https://github.com/nextgenusfs/{{ name }}/archive/{{ version }}.tar.gz
-  sha256: ba8fb8ba0bb440863031613b7ce2074ab356abbdf5931280e582261f4a0e30cc
+  sha256: 54c109237fc06bda36bb788b5ae1069593171f4a36db037a078e6736e04292cc
```
Comment (bgruening) [inline recipes/amptk/meta.yaml]: why has the checksum changed?

Comment (nextgenusfs) [reply recipes/amptk/meta.yaml]: I had a slight typo in code.

File: recipes/amptk/meta.yaml [hunk 2]
```diff
@@ -40,3 +40,7 @@ about:
   license_file: LICENSE.md
   summary: 'AMPtk: Amplicon tool kit for processing high throughput amplicon sequencing data.'
   license_family: BSD
+
```
Comment (bgruening) [inline recipes/amptk/meta.yaml]: Thanks for adding the DOI. If you only add this you don't need to bump the build number. I'm wondering why you needed to change the checksum.

Comment (nextgenusfs) [reply recipes/amptk/meta.yaml]: source changed slightly due to a fixed typo, if bumping the version number is better I can do that as well.

File: recipes/amptk/meta.yaml (hunk 1 above)
Comment (bgruening) [reply recipes/amptk/meta.yaml]: This is not good practice. You don't change the version, but in the results can change. Better is to do released and minor releases.

Comment (bgruening) [reply recipes/amptk/meta.yaml]: If you decrease the build number again to 0, I will remove the previous version from anaconda.org

**Pull request merged**